script.js

```diff
@@ -1,4 +1,443 @@
+//states
+var playerCount = 0;
+var userNames = [];
+var gameMode = "";
+var gameType = "";
+var diceCount = 0;
+var playerScores = [];
+var diceRolls = [];
+var playGameCount = 0;
+var scoreBoard = [];
+var tempScoreBoard = [];
+var gameRoundCount = 0;
+var eliminationArray = [];
+
+//constants
+const NORMAL = "Normal";
+const LOWEST = "Lowest";
+const KNOCKOUT = "Knockout";
+const HIGHSCORE = "Highscore";
+const DICE = [1, 2, 3, 4, 5, 6];
+
+//username change state
+var userNameCount = 11;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // player count validation
+  if (playerCount == 0) {
+    if (isNaN(input) || input > 10 || input < 2) {
+      return `Please enter the number of players that wish to play. Maximum of 10 and minimum of 2!`;
+    }
+    playerCount = Number(input);
+    // update scoreboard
+    for (let a = 0; scoreBoard.length < playerCount; a += 1) {
+      scoreBoard.push(Number(0));
+    }
+
+    console.log("scoreboard fill 0: " + scoreBoard);
+    return `<b>${playerCount}</b> players selected. Please proceed to enter your usernames.`;
+  }
+
+  //username validation
+  if (userNames.length < playerCount) {
+    if (input == "" || userNames.includes(input)) {
+      return `Please enter a valid username! Usernames must be unique to other players and should not be blank!`;
+    } else if (userNames.length == playerCount - 1 && diceCount != 0) {
+      userNames.push(input);
+      console.log(userNames);
+      return `Hello <b>${input}</b>, Welcome to the game. We now have your names 😈 Click submit to continue the game`;
+    } else if (userNames.length == playerCount - 1) {
+      userNames.push(input);
+      console.log(userNames);
+      return `Hello <b>${input}</b>, Welcome to the game. We now have your names 😈 Please enter the number of dices you want to play with.`;
+    }
+    userNames.push(input);
+    return `Hello <b>${input}</b>, Welcome to the game.`;
+  }
+
+  //diceCount validation
+  if (diceCount == 0) {
+    if (isNaN(input) || input > 10 || input < 2) {
+      return `Please enter the number of dices you wish to play with. Maximum of 10 and minimum of 2!`;
+    } //for changing dice count
+    else if (gameMode != "") {
+      diceCount = Number(input);
+      return `Dice count successfully changed to <b>${diceCount}</b>. Click submit to continue rolling!`;
+    }
+    diceCount = Number(input);
+    return `<b>${diceCount}</b> dices selected. Please proceed to select your game mode.<br><br>Modes available:<br>⮞Highscore<br>⮞Knockout`;
+  }
+
+  //gameMode validation
+  if (gameMode == "") {
+    if (
+      input == "" ||
+      (input.toUpperCase() != HIGHSCORE.toUpperCase() &&
```

Comment:
Good practice of changing the case of the input so that user errors can't break your code. Can you think of how we can transform input into uppercase in just one line, rather than every instance that we use input? (e.g. in line 76)